lib/cartopy/io/ogc_clients.py

```diff
@@ -232,8 +232,8 @@ def _image_and_extent(self, wms_proj, wms_srs, wms_extent, output_proj,
         return LocatedImage(wms_image, extent)
 
     def fetch_raster(self, projection, extent, target_resolution):
-        service = self.service
         min_x, max_x, min_y, max_y = extent
+        target_resolution = [np.ceil(val) for val in target_resolution]
```

Comment:
Ditto to all of the above. I took @ocefpaf's line of thinking: if you want an image of 300.1 pixels, did you actually mean 301 or 300? These numbers are computed automatically based on dpi and figure size, so I went for the approach of giving a little more, rather than a little less, though in practice, it makes virtually no difference at all.